src/main/java/kr/codesqaud/cafe/controller/ArticleController.java

```diff
@@ -22,23 +20,41 @@ public ArticleController(ArticleService articleService) {
         this.articleService = articleService;
     }
 
-    @GetMapping
-    public String showPostArticleForm(){
-        return "post/form";
-    }
-
-    @PostMapping
-    public String postArticle(@Valid ArticleDTO articleDto, BindingResult bindingResult) {
-        if(bindingResult.hasErrors()){
-            return "post/form";
-        }
+    @PostMapping("/article/submit")
+    public String postArticle(@ModelAttribute @Valid ArticleDTO articleDto, HttpSession httpSession) {
+        Session session = (Session) httpSession.getAttribute(Session.LOGIN_USER);
+        articleDto.setId(session.getId());
+        articleDto.setNickName(session.getNickName());
         articleService.post(articleDto);
         return "redirect:/";
     }
 
-    @GetMapping("/{id}")
-    public String showDetailArticle(@PathVariable int id, Model model) {
-        model.addAttribute("article", articleService.findArticleById(id));
+    @GetMapping("/article/{idx}")
+    public String showDetailArticle(@PathVariable long idx, Model model) {
+        model.addAttribute("article", articleService.findArticleByIdx(idx));
         return "post/show";
     }
+
+    @GetMapping("/article/update-form/{idx}")
+    public String showUpdateForm(@PathVariable Long idx,Model model,HttpSession httpSession){
```

Comment:
```suggestion
    public String showUpdateForm(@PathVariable Long idx,Model model,HttpSession httpSession) {
```
code formatting이 되지 않았군요. naver campus intellij formatting 사용을 고려해보세요.
https://naver.github.io/hackday-conventions-java/

Reply:
알려주신 사이트 참고하여 코딩 컨벤션 적용했습니다.